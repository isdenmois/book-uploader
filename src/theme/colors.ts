export const primary = '#000';
export const secondary = '#eee';
export const invert = '#fff';

export const green = '#8BC34A';
export const red = '#f44336';
export const darkBlue = '#007ac1';
export const blue = '#2196f3';
export const lightBlue = '#67daff';

export const Background = '#F9FAFE';
export const SearchBackground = '#B8D5F9';
export const UploadBackground = '#C2B9F9';
export const ProfileBackground = '#FFAD99';
export const TabsBackground = 'white';

export const SearchText = '#092D5D';
export const UploadText = '#15085E';
export const ProfileText = '#661400';
export const TabText = '#121417';

export const SearchSelected = '#5396EE';
export const UploadSelected = '#6851F0';
export const ProfileSelected = '#FF6842';

export const Secondary = '#787D87';
export const Search = '#8C93A3';
export const Input = 'white';
export const DeleteButton = '#9BA8CA';
export const Success = '#48B755';
export const Error = '#FD4463';
