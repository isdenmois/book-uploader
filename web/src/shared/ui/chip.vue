<script setup lang="ts">
defineProps<{
  title: string
  selected?: boolean
  disabled?: boolean
}>()
const emit = defineEmits<{ (e: 'select'): void }>()
</script>

<template>
  <button class="chip" :class="{ selected }" :disabled="disabled" @click="emit('select')">{{ title }}</button>
</template>

<style lang="scss">
.chip {
  border: 1px solid var(--secondaryBackground);
  border-radius: 20px;
  color: var(--secondary);
  cursor: pointer;
  margin-left: 8px;
  padding: 4px 12px;
  font-size: 14px;

  &.selected {
    background-color: var(--searchBackground);
    border-color: var(--searchSelected);
    color: var(--searchText);
  }
}
</style>
