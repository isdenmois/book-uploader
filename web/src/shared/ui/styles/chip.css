.chip {
  border: 1px solid var(--secondaryBackground);
  border-radius: 20px;
  color: var(--secondary);
  cursor: pointer;
  margin-left: 8px;
  padding: 4px 12px;
  font-size: 14px;
}

.chip.selected {
  background-color: var(--searchBackground);
  border-color: var(--searchSelected);
  color: var(--searchText);
}
