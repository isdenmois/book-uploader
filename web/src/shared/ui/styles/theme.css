button,
input {
  all: unset;
}

button:focus {
  outline: none;
}

body {
  background-color: var(--background);
}

#app {
  max-width: 600px;
  margin: 0 auto;
  height: 100vh;
  overflow: hidden;
  display: flex;
  flex-direction: column;
}

h1 {
  font-size: 24px;
  color: var(--profileText);
}

.book-list > div:not(:first-child) {
  @apply mt-4;
}

.input {
  background: #ffffff;
  box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.25);
  border-radius: 10px;
  padding-left: 4px;
}
