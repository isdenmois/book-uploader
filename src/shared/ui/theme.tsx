import React, { FC } from 'react'
import { TouchableOpacity, TouchableOpacityProps, useColorScheme } from 'react-native'
import {
  ThemeProvider as ReThemeProvider,
  createBox,
  createText,
  createTheme,
  useTheme as useThemeRN,
} from '@shopify/restyle'

import * as Colors from './colors'

export const theme = createTheme({
  spacing: {
    0.25: 2,
    0.5: 4,
    1: 8,
    2: 16,
    3: 24,
    4: 32,
    5: 40,
    6: 48,
  },
  colors: Colors.light,
  breakpoints: {},
  textVariants: {
    defaults: {
      fontSize: 16,
    },
    chip: {
      fontSize: 14,
      lineHeight: 18,
    },
    button_primary: {
      fontSize: 16,
      lineHeight: 22,
      color: 'invertedText',
    },
  },
  buttonVariants: {
    primary: {
      backgroundColor: 'profileSelected',
    },
  },
})

export type Theme = typeof theme

const darkTheme: Theme = {
  ...theme,
  colors: {
    ...theme.colors,
    ...Colors.dark,
  },
}

export const Box = createBox<Theme>()
export const Text = createText<Theme>()
export const TouchableBox = createBox<Theme, TouchableOpacityProps & { children: any }>(TouchableOpacity)

export const ThemeProvider: FC = ({ children }) => {
  const currentMode = useColorScheme()

  return <ReThemeProvider theme={currentMode === 'dark' ? darkTheme : theme}>{children}</ReThemeProvider>
}

export const useTheme = () => useThemeRN<Theme>()
