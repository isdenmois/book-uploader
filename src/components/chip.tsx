import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import { DynamicStyleSheet, useDynamicStyleSheet } from 'react-native-dynamic';
import { dynamicColor } from 'theme/colors';

type Props = {
  title: string;
  selected: boolean;
  onPress: () => void;
  disabled?: boolean;
};

export function Chip({ title, selected, disabled, onPress }: Props) {
  const s = useDynamicStyleSheet(ds);

  return (
    <TouchableOpacity
      style={selected ? [s.container, s.selected] : s.container}
      disabled={disabled}
      onPress={onPress}
      testID={title}
    >
      <Text style={selected ? [s.text, s.selectedText] : s.text} testID={selected ? 'selected' : null}>
        {title}
      </Text>
    </TouchableOpacity>
  );
}

const ds = new DynamicStyleSheet({
  container: {
    paddingHorizontal: 15,
    paddingVertical: 8,
    borderRadius: 20,
    borderWidth: 1,
    borderColor: dynamicColor.secondaryBackground,
    marginLeft: 15,
  },
  text: {
    color: dynamicColor.secondary,
    height: 18,
    fontSize: 14,
  },
  selected: {
    backgroundColor: dynamicColor.searchBackground,
    borderColor: dynamicColor.searchSelected,
  },
  selectedText: {
    color: dynamicColor.searchText,
  },
});
