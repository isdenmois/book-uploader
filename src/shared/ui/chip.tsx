import React, { FC } from 'react'
import { Text, TouchableBox } from 'shared/ui/theme'

type Props = {
  title: string
  selected: boolean
  onPress: () => void
  disabled?: boolean
}

export const Chip: FC<Props> = ({ title, selected, disabled, onPress }) => {
  return (
    <TouchableBox
      px={2}
      py={1}
      ml={2}
      borderColor={selected ? 'searchSelected' : 'secondaryBackground'}
      backgroundColor={selected ? 'searchBackground' : null}
      borderRadius={20}
      borderWidth={1}
      disabled={disabled}
      onPress={onPress}
      testID={title}
    >
      <Text variant='chip' color={selected ? 'searchText' : 'secondary'} testID={selected ? 'selected' : null}>
        {title}
      </Text>
    </TouchableBox>
  )
}
