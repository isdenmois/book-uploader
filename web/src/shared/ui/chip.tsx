import { Component } from 'solid-js'
import './styles/chip.css'

interface Props {
  title: string
  selected?: boolean
  disabled?: boolean
  onSelect: () => void
}

export const Chip: Component<Props> = props => {
  return (
    <button class='chip' classList={{ selected: props.selected }} disabled={props.disabled} onClick={props.onSelect}>
      {props.title}
    </button>
  )
}
