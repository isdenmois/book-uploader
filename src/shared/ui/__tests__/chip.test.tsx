import React from 'react'
import { act, render } from '@testing-library/react-native'
import { ThemeProvider } from '../theme'
import { Chip } from '../chip'

test('Chip', async () => {
  const { toJSON, rerender, unmount } = render(<Chip title='Var1' />, { wrapper: ThemeProvider })

  expect(toJSON()).toMatchSnapshot()

  rerender(<Chip title='Var1' selected />)
  expect(toJSON()).toMatchSnapshot()

  await act(async () => {
    unmount()
  })
})
