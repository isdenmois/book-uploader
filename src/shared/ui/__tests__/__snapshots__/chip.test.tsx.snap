// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": null,
      "borderColor": "#E0E0E0",
      "borderRadius": 20,
      "borderWidth": 1,
      "display": "flex",
      "marginLeft": 16,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
  testID="Var1"
>
  <Text
    style={
      Array [
        Object {
          "color": "#787D87",
          "display": "flex",
          "fontSize": 14,
          "lineHeight": 18,
        },
      ]
    }
    testID={null}
  >
    Var1
  </Text>
</View>
`;

exports[`Chip 2`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#B8D5F9",
      "borderColor": "#5396EE",
      "borderRadius": 20,
      "borderWidth": 1,
      "display": "flex",
      "marginLeft": 16,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
  testID="Var1"
>
  <Text
    style={
      Array [
        Object {
          "color": "#092D5D",
          "display": "flex",
          "fontSize": 14,
          "lineHeight": 18,
        },
      ]
    }
    testID="selected"
  >
    Var1
  </Text>
</View>
`;
